parsers/rwcpopmelody_parser.py

```diff
@@ -0,0 +1,197 @@
+
+The original dataset is described online at the following URL:
+    https://staff.aist.go.jp/m.goto/RWC-MDB/rwc-mdb-p.html
+
+To parse the entire dataset, you just need to provide the path to the folder
+containing the SMF files, with filenames in the form RM-P[0-9]{3}*.MID from
+RM-P001*.MID to RM-P100*.MID
+
+Example:
+./rwcpopmelody_parser.py ~/AIST.RWC-MDB-P-2001.SMF_SYNC -o ~/RWCPOP_melody_jams/
+
+"""
+
+__author__ = "J. Salamon"
+__copyright__ = "Copyright 2016, Music and Audio Research Lab (MARL)"
+__license__ = "GPL"
+__version__ = "1.0"
+__email__ = "justin.salamon@nyu.edu"
+
+import argparse
+import logging
+import os
+import time
+import midi
+import pretty_midi
+import pandas as pd
+
+import jams
+
+
+def fill_file_metadata(jam, metadata, n_track):
+    """Fills the global metada into the JAMS jam."""
+    jam.file_metadata.artist = metadata['Artist (Vocal)'][n_track]
+    jam.file_metadata.title = metadata['Title'][n_track]
+
+    d_str = metadata['Length'][n_track]
+    jam.file_metadata.duration = (float(d_str.split(":")[0]) * 60 +
+                                  float(d_str.split(":")[1]))
+
+    # Store remaining RWC metadata in the sandbox
+    sandbox_dict = {
+        'Piece No.': metadata['Piece No.'][n_track],
+        'Cat. Suffix': metadata['Cat. Suffix'][n_track],
+        'Tr. No.': metadata['Tr. No.'][n_track],
+        'Singer Information': metadata['Singer Information'][n_track],
+        'Tempo': metadata['Tempo'][n_track],
+        'Live Instruments Used': metadata['Live Instruments Used'][n_track],
+        'Drum Information': metadata['Drum Information'][n_track]}
+
+    jam.sandbox.update(**sandbox_dict)
+
+
+def fill_annotation_metadata(annot):
+    """Fills the annotation metadata."""
+    annot.annotation_metadata.corpus = "RWC Music Database: Popular Music"
+    annot.annotation_metadata.version = "1.0"
+    annot.annotation_metadata.annotation_tools = ""
+    annot.annotation_metadata.annotation_rules = ""
+    annot.annotation_metadata.validation = ""
+    annot.annotation_metadata.data_source = ""
+    annot.annotation_metadata.curator = jams.Curator(name="Masataka Goto",
+                                                     email="m.goto@aist.go.jp")
+    annot.annotation_metadata.annotator = {}
+
+
+def create_jams(smf_file, out_file, metadata):
+    """
+    Creates a JAMS file from an RWC POP smf file (RM-P*.MID).
+    Note: only the notes of the MELODY track are kept!
+    """
+
+    # Load midi file
+    m = midi.read_midifile(smf_file)
+
+    # This will store the relevant MIDI tracks
+    melody = []
+
+    # Track 0 is metadata that we need
+    melody.append(m[0])
+
+    # Collect track text (convert to lower case and remove spaces)
+    track_text = ["".join(e.text.lower().split()) for t in m
+                  for e in t if isinstance(e, midi.TrackNameEvent)]
+
+    assert len(track_text) == len(m)
+
+    # Find the melody track: it should START with "melo", or if that's not
```

Comment:
Is this a standard convention?  Or would it make sense to abstract this out and let a user specify a regular expression to search for instead?  I like `'/melo|voca/'` more than hard-wired logic here.